src/App.tsx

```diff
@@ -1,27 +1,68 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handleClickAll = () => {
+    const loadGoods = async () => {
+      const goodsData = await getAll();
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+      setGoods(goodsData);
+    };
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+    loadGoods();
+  };
 
-    <GoodsList goods={[]} />
-  </div>
-);
+  const handleClickFirst5 = () => {
+    const loadGoods = async () => {
```

Comment:
Consider wrapping the async function `loadGoods` in a `useCallback` hook to avoid unnecessary re-renders.

Reply:
Or maybe you can make handleClickFirst5 async and avoid creating a function in a function